CMakePresets.json

```diff
@@ -0,0 +1,309 @@
+{
```

Comment:
Is there a known way to override settings in this file without touching files tracked by git? If not, are we sure that enforcing them is ok, i.e. there is no possible reason to change anything?

Reply:
> Is there a known way to override settings in this file without touching files tracked by git?

Not that I know of. It's the same problem you mentioned for the vscode folder. Git is going to pick up the change if you modify it.
However when running cmake manually you have to explicitly use this file `cmake --preset <preset-name>`. Not sure how it works for other IDEs.

Reply:
If there is no ability to override it without touching and there is valid reason to override it, it should not be tracked in the repository. It causes problems, not only for me: https://github.com/FreeCAD/FreeCAD/pull/11138#issuecomment-1773964099. 

This PR excludes `settings.json` from the repository (big yay) - maybe there is VS Code setting which allows users to provide path to custom preset? That'd solve the issue.

Reply:
OK, CMake authors are good devs: https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#id3

> `CMakePresets.json` and `CMakeUserPresets.json` live in the project's root directory. They both have exactly the same format, and both are optional (though at least one must be present if [--preset](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-preset) is specified). `CMakePresets.json` is meant to specify project-wide build details, while `CMakeUserPresets.json` is meant for developers to specify their own local build details.
> 
> `CMakePresets.json` may be checked into a version control system, and `CMakeUserPresets.json` should NOT be checked in. For example, if a project is using Git, `CMakePresets.json` may be tracked, and `CMakeUserPresets.json` should be added to the `.gitignore`.

That basically solves this issue.

Reply:
> That basically solves this issue, but we need to add CMakeUserPresets.json to .gitignore, @oursland can you do that?

Alright, this allows for individual customization, but cmakepresets might still interfere with IDEs. having the user copy it manually (like the vscode folder) ensures there will be no issues

Reply:
I think that if IDEs support the `CMakePresets.json` they support he override too, but it should be checked.